src/vcs.rs

```diff
@@ -25,6 +25,19 @@ impl PlatformId {
     }
 }
 
+/// Generic definition of an online Git project.
+pub trait GitProject {
```

Comment:
I'd recommend rather calling this `GenericProject`, and renaming the `OnlineGitProject` enum to `Project`.

Both are members of the `vcs` package, so I think it's clear that they're VCS projects.